### PR TITLE
added Movie, CinemaHall and MovieSession controllers

### PR DESCRIPTION
**Pull request by entertainment-veks**
changed their services and daos

### CODE REVIEW

File: src/main/java/hibernate/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import hibernate.service.mapper.CinemaHallMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (Fedich09) [inline src/main/java/hibernate/controller/CinemaHallController.java:16]:
```suggestion
@RestController
@RequestMapping("/cinema-halls")
```

File: src/main/java/hibernate/controller/CinemaHallController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private CinemaHallMapper cinemaHallMapper;
+
+    @Autowired
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @PostMapping("/cinema-halls")
```
Comment (Fedich09) [inline src/main/java/hibernate/controller/CinemaHallController.java:28]:
```suggestion
    @PostMapping
```

File: src/main/java/hibernate/controller/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public void saveCinemaHall(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
+        CinemaHall current = cinemaHallMapper.parseFromDto(cinemaHallRequestDto);
+        cinemaHallService.add(current);
+    }
+
+    @GetMapping("/cinema-halls")
```
Comment (Fedich09) [inline src/main/java/hibernate/controller/CinemaHallController.java:34]:
```suggestion
    @GetMapping
```

File: src/main/java/hibernate/controller/MovieController.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import hibernate.service.mapper.MovieMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (Fedich09) [inline src/main/java/hibernate/controller/MovieController.java:16]: same here
```suggestion
@RestController
```

File: src/main/java/hibernate/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (Fedich09) [inline src/main/java/hibernate/controller/MovieSessionController.java:22]:
```suggestion
@RestController
```
and here

File: src/main/java/hibernate/dao/impl/CinemaHallDaoImpl.java [hunk 6]
```diff
@@ -51,4 +51,16 @@ public CinemaHall add(CinemaHall cinemaHall) {
...
         }
     }
+
+    @Override
+    public CinemaHall get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<CinemaHall> query = session.createQuery("from CinemaHall"
+                    + " WHERE id = :id", CinemaHall.class);
+            query.setParameter("id", id);
+            return query.getSingleResult();
```
Comment (Fedich09) [inline src/main/java/hibernate/dao/impl/CinemaHallDaoImpl.java:61]:
```suggestion
           return session.get(CinemaHall.class, id);
```
you can use this method

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 7]
```diff
@@ -53,4 +54,16 @@ public Movie add(Movie movie) {
             throw new DataProcessingException("Can't get all", e);
         }
     }
+
+    @Override
+    public Movie get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Movie> query = session.createQuery("from Movie"
```
Comment (Fedich09) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:61]: same here, but I do not consider it a mistake

**Review by Fedich09 — CHANGES_REQUESTED** [review]

**Review by Fedich09 — APPROVED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**